.github/workflows/build_and_test.yml

```diff
@@ -8,28 +8,127 @@ on:
 
-      - name: Install Node.js
+      - name: Install Node
         uses: actions/setup-node@v2.2.0
         with:
           node-version: 16
           cache: npm
       
-      - name: Install dependencies
+      - name: Install NPM Dependencies
         run: npm ci
 
-      - name: Build web app
+      - name: Build Web App
         run: npx gulp build --platform=browser
-      
-      - name: Test web app
+
+      - name: Test Web App
         run: npm run action scripts/test
 
-      - name: Build Electron app
-        run: npm run action src/electron/build
+      - name: Build Electron App
+        run: npm run action src/electron/package_linux
+
+  windows:
+    name: Windows
+    runs-on: windows-2019
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+
+      - name: Install Node
+        uses: actions/setup-node@v2.2.0
+        with:
+          node-version: 16
+          cache: npm
+
+      - name: Install NPM Dependencies
+        run: npm ci
+
+      - name: Build Windows Client
+        run: npm run action src/electron/package_windows
+
+  mac:
+    name: MacOS
+    runs-on: macos-11
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+
+      - name: Install Node
+        uses: actions/setup-node@v2.2.0
+        with:
+          node-version: 16
+          cache: npm
+
+      - name: Install NPM Dependencies
+        run: npm ci
+
+      - name: Set XCode Version
+        run: sudo xcode-select -switch /Applications/Xcode_12.5.app
+
+      - name: Build OSX Client
+        run: npx gulp build --platform=osx
+
+  # TODO: setup fastlane for iOS
+  
+  # ios:
+  #   name: iOS
+  #   runs-on: macos-11
+  #   steps:
+  #     - name: Checkout
+  #       uses: actions/checkout@v2.3.4
+
+  #     - name: Install Node
+  #       uses: actions/setup-node@v2.2.0
+  #       with:
+  #         node-version: 16
+  #         cache: npm
+
+  #     - name: Install NPM Dependencies
+  #       run: npm ci
+
+  #     - name: Set XCode Version
+  #       run: sudo xcode-select -switch /Applications/Xcode_12.5.app
+
+  #     - name: Build iOS Client
+  #       run: npm run action apple/scripts/ios_package_remote
+
+  android:
+    name: Android
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+
+      - name: Install Node
+        uses: actions/setup-node@v2.2.0
+        with:
+          node-version: 16
+          cache: npm
+
+      - name: Install NPM Dependencies
+        run: npm ci
+
+      - name: Install Java
+        uses: actions/setup-java@v1.4.3
+        with:
+          java-version: 1.8
+
+      - name: Setup Android
+        uses: android-actions/setup-android@v2
+
+      # - name: Set Android SDK Version
+      #   run: |
+      #     sdkmanager "build-tools;30.0.2" "platforms;android-29"
+
+      - name: Build Android Client
+        run: npx gulp build --platform=android
+        env:
+          ANDROID_BUILD_TOOLS_VERSION: 30.0.2
```

Comment:
Where's the install step?

Reply:
I'm currently using `android-actions/setup-android@v2`!  https://github.com/Jigsaw-Code/outline-client/pull/1096/files/99f96efba2ecec187c2057c245acfb6eac04fea4#diff-48c0ee97c53013d18d6bbae44648f7fab9af2e0bf5b0dc1ca761e18ec5c478f2R123-R124

The tricky thing is accepting the license after downloading `sdkmanager`, so I feel using this is easier to maintain than trying to figure out how to do it ourselves.

However, what I'm currently trying to figure out is how to set the `build-tools` without explicitly uninstalling v31 - i've tried `sdkmanager "build-tools;30.0.2"` and also setting the env here, nothing quite sticks. Putting this PR back into draft while I figure it out.

Reply:
It's easy to accept the license by piping `yes` into sdkmanager:
https://github.com/Jigsaw-Code/outline-client/blob/4e62987183e36ddb42d6dbbcea4cac19d013adf1/tools/build/Dockerfile#L60